Lifetime in type signature of Table::get is more restrictive than it needs to be [Rust 0.4.0, flatbuffers master]

- Language: Rust
- Rust crate version: 0.4.0
- Rust version: latest nightly

Logically, because the `Table` struct is simply a reference, this code should be able to compile:

```rust
extern crate flatbuffers;

use flatbuffers::Table;

static DATA: [u8; 3] = [1, 2, 3]; // some binary data

fn main() {
    let table: Table<'static> = Table::new(&DATA, 0);
    table.get::<&'static str>(0, None);
}
```

However, it generates a compiler error:

```
error[E0597]: `table` does not live long enough
  --> src/main.rs:9:5
   |
9  |     table.get::<&'static str>(0, None);
   |     ^^^^^ borrowed value does not live long enough
10 | }
   | - borrowed value only lives until here
   |
   = note: borrowed value must be valid for the static lifetime...
```

This is because of the type signature of `Table::get`:

```rust
pub fn get<T: Follow<'a> + 'a>(
    &'a self,
    slot_byte_loc: VOffsetT,
    default: Option<T::Inner>
) -> Option<T::Inner>
```

The `'a` in `&'a self` is unnecessary. It should simply be `&self`.

--- a/rust/flatbuffers/src/table.rs
+++ b/rust/flatbuffers/src/table.rs
@@ -30,12 +30,12 @@ impl<'a> Table<'a> {
         Table { buf: buf, loc: loc }
     }
     #[inline]
-    pub fn vtable(&'a self) -> VTable<'a> {
+    pub fn vtable(&self) -> VTable<'a> {
         <BackwardsSOffset<VTable<'a>>>::follow(self.buf, self.loc)
     }
     #[inline]
     pub fn get<T: Follow<'a> + 'a>(
-        &'a self,
+        &self,
         slot_byte_loc: VOffsetT,
         default: Option<T::Inner>,
     ) -> Option<T::Inner> {

--- a/tests/rust_usage_test/tests/integration_test.rs
+++ b/tests/rust_usage_test/tests/integration_test.rs
@@ -236,6 +236,37 @@ mod generated_constants {
 }
 
 #[cfg(test)]
+mod lifetime_correctness {
+    extern crate flatbuffers;
+
+    use std::mem;
+
+    use super::my_game;
+    use super::load_file;
+
+    #[test]
+    fn table_get_field_from_static_buffer_1() {
+        let buf = load_file("../monsterdata_test.mon");
+        // create 'static slice
+        let slice: &[u8] = &buf;
+        let slice: &'static [u8] = unsafe { mem::transmute(slice) };
+        // make sure values retrieved from the 'static buffer are themselves 'static
+        let monster: my_game::example::Monster<'static> = my_game::example::get_root_as_monster(slice);
+        // this line should compile:
+        let name: Option<&'static str> = monster._tab.get::<flatbuffers::ForwardsUOffset<&str>>(my_game::example::Monster::VT_NAME, None);
+        assert_eq!(name, Some("MyMonster"));
+    }
+
+    #[test]
+    fn table_get_field_from_static_buffer_2() {
+        static DATA: [u8; 4] = [0, 0, 0, 0]; // some binary data
+        let table: flatbuffers::Table<'static> = flatbuffers::Table::new(&DATA, 0);
+        // this line should compile:
+        table.get::<&'static str>(0, None);
+    }
+}
+
+#[cfg(test)]
 mod roundtrip_generated_code {
     extern crate flatbuffers;
 

